### PR TITLE
Ods loses spaces 32207

### LINKED ISSUE #32207: read_excel loses spaces on ods
#### Code Sample
Create a new spreadsheet with 1 column "testcol" in LibreOffice / OpenOffice & Excel, save as `test.ods`/`test.xlsx`:
```
testcol
this is great
4    spaces
1 trailing 
 1 leading
2  spaces  multiple  times
```
For simplicity here as zip: (1 ods, 1 xlsx): [spreadsheets.zip](https://github.com/pandas-dev/pandas/files/4242180/spreadsheets.zip)

```python
import pandas as pd
for i in pd.read_excel('test.ods', engine='odf')['testcol']:
    print(repr(i))

# output:
'this is great'
'4 spaces'
'1 trailing '
'1 leading'
'2 spaces multiple times'

for i in pd.read_excel('test.xlsx')['testcol']:
    print(repr(i))

# output:
'this is great'
'4    spaces'
'1 trailing '
' 1 leading'
'2  spaces  multiple  times'
```

#### Problem description
When reading `.ods` files (OpenOffice or LibreOffice) multiple spaces are collapsed into one, leading ones are lost, trailing ones preserved.

#### Expected Output
see excel output above.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.6.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 17.7.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : en_US.UTF-8
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.0.1
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 45.2.0
Cython           : 0.29.15
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.5.0
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.12.0
pandas_datareader: None
bs4              : 4.8.2
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.5.0
matplotlib       : 3.1.3
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : 1.2.0
xlwt             : None
xlsxwriter       : None
numba            : 0.48.0
</details>

Not sure why that says `odfpy: None`, `pip list` tells me it's 1.4.1

### PR DESCRIPTION
**Pull request by detrout**
- [X] closes #32207
- [X] tests added / passed tests.io.excel.test_reader_spaces
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff` (I had to use git diff master though)
- [ ] whatsnew entry. It's only a bug fix does it need a whatsnew?

I'm not sure how you make xlsxb files so that test wasn't implement when checking for the different types of space.

There's quite possibly many other ways this parser doesn't handle the odf specification.

### CODE REVIEW

File: pandas/io/excel/_odfreader.py [hunk 1]
```diff
@@ -182,3 +182,24 @@ def _get_cell_value(self, cell, convert_float: bool) -> Scalar:
             return pd.to_datetime(str(cell)).time()
         else:
             raise ValueError(f"Unrecognized type {cell_type}")
+
+    def _get_cell_string_value(self, cell):
+        from odf.element import Text, Element
```
Comment (WillAyd) [inline pandas/io/excel/_odfreader.py:187]: Can you just add a quick docstring as to why this required? i.e. that it preserves spaces

**Review by WillAyd — COMMENTED** [review]
just pushed an .xlsb copy up; did it with a "Save As" from microsoft excel not sure if supported by other tools

**Comment by WillAyd** [comment]
Can you also add a whatsnew for 1.1?

**Comment by WillAyd** [comment]
Can you run black on the changes? Should fix the code checks failure

The linux 37 failure is probably unrelated

**Comment by detrout** [comment]
While I was filling out the form I ran black, but forgot to push the commit.

And thanks for uploading a xlsb file, I don't have a copy of excel here at home.

**Comment by WillAyd** [comment]
If you can add a whatsnew note for 1.1 this lgtm - thanks!

**Comment by detrout** [comment]
How's the description?

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
@jreback care to look?

**Comment by detrout** [comment]
> looks ok, i imagine this would slow down writing a bit?

There's no OpenDocument write support so I don't think it can slow down writes.
I'll need to figure out how to have a stable test environment for running a benchmark

**Comment by jreback** [comment]
thanks @detrout

**Pull request merged**